### PR TITLE
Stable Release

### PR DESCRIPTION
**Pull request by AminovE99**
:sparkles: Stable Release including LDAP and website signup configuration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+
+## How to use
+
+Environment variables for automatic setup:
+* WebSite configuration
+  * **TODO**
+* Email configuration
+  * **TODO**
+* LDAP configuration
+  * **TODO**
```
Comment (madmath03) [inline README.md]: @AminovE99 Need to list the available environment variables.

File: test/docker-compose.mariadb.yml [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      - ENCRYPTION_KEY=${ERPNEXT_ENCRYPTION_KEY}
+      # Database setup
+      - DB_TYPE=mariadb
+      - DB_HOST=erpnext_db
+      - DB_PORT=3306
+      - DB_NAME=${ERPNEXT_DB_NAME}
+      - DB_PASSWORD=${ERPNEXT_DB_PWD}
+      - DB_ROOT_PASSWORD=${ERPNEXT_DB_ROOT_PWD}
+      # Mail setup
+      - MAIL_MUTED=true
```
Comment (madmath03) [inline test/docker-compose.mariadb.yml:74]: @AminovE99 We need to add the environment values to check that the autoinstall works when configuration is defined.
We might need to define a second docker-compose file (one with the config and one without).

File: erpnext_autoinstall/install.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+#Todo: maybe should remade as @decorator
+def configure_ldap():
+    if os.getenv('LDAP_SERVER_URL') and os.getenv('LDAP_BASE_DN') and os.getenv('LDAP_PASSWORD') and os.getenv(
+            'LDAP_USERS_ORGANIZATIONAL_UNIT') and os.getenv('LDAP_DEFAULT_ROLE', 'Employee') and os.getenv(
+        'LDAP_SEARCH_STRING', 'uid=\{0\}') and os.getenv('LDAP_EMAIL', 'mail') and os.getenv('LDAP_USERNAME',
+                                                                                             'uid') and os.getenv(
+        'LDAP_FIRST_NAME', 'givenName') and os.getenv('LDAP_LAST_NAME', 'sn') and os.getenv('LDAP_PHONE_FIELD',
+                                                                                            'telephoneNumber') and os.getenv(
+        'LDAP_MOBILE_FIELD', 'mobile') and os.getenv('LDAP_SSL_TLS_MODE'):
+        doc = frappe.get_doc("LDAP Settings")
```
Comment (madmath03) [inline erpnext_autoinstall/install.py]: @AminovE99 You do not need to test values that have a a default value ;-)

File: erpnext_autoinstall/install.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+def disable_registration():
+    doc = frappe.get_doc("Website Settings")
+    doc.disable_signup = int(os.getenv('DISABLE_SIGNUP', '0'))
+    doc.save()
+
+
+def configure_domain():
+    if os.getenv('EMAIL SERVER') and os.getenv('EMAIL_ADDRESS') and os.getenv('EMAIL_DOMAIN_NAME') and os.getenv(
+            'EMAIL_SMTP_SERVER') and os.getenv('EMAIL_PORT') and os.getenv('EMAIL_USE_IMAP') and os.getenv(
+        'EMAIL_USE_SSL') and int(os.getenv('EMAIL_USE_TLS')) and os.getenv('EMAIL_ATTACHMENT_LIMIT_MB'):
```
Comment (madmath03) [inline erpnext_autoinstall/install.py]: @AminovE99 We should probably add default values to the most common email properties (port 993 and use SSL true for instance)

**Review by madmath03 — CHANGES_REQUESTED** [review]
@AminovE99 A few things to add in tests and CI.

File: README.md (hunk 1 above)
Comment (AminovE99) [reply README.md]: Done, I think

File: erpnext_autoinstall/install.py (hunk 4 above)
Comment (AminovE99) [reply erpnext_autoinstall/install.py]: Done

File: README.md (hunk 1 above)
Comment (madmath03) [reply README.md]: @AminovE99 nope, you added the environment variables in `.env` file, but not in the `README.md` documentation.

File: erpnext_autoinstall/install.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import frappe
+
+
+def after_install():
+    disable_registration()
+    print("REGISTRATION HAS BEEN DISABLED")
+    configure_email()
+    print("EMAIL HAS BEEN CONFIGURED")
+    configure_ldap()
+    print("LDAP HAS BEEN CONFIGURED!")
```
Comment (madmath03) [inline erpnext_autoinstall/install.py]: @AminovE99 Good idea to print what configuration has been done, but it should be inside each function, so if configuration has been actually done.

File: erpnext_autoinstall/install.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        doc.ldap_email_field = os.getenv('LDAP_EMAIL', 'mail')
+        doc.ldap_username_field = os.getenv('LDAP_USERNAME', 'uid')
+        doc.ldap_first_name_field = os.getenv('LDAP_FIRST_NAME', 'givenName')
+        doc.ldap_last_name_field = os.getenv('LDAP_LAST_NAME', 'sn')
+        doc.ldap_phone_field = os.getenv('LDAP_PHONE_FIELD', 'telephoneNumber')
+        doc.ldap_mobile_field = os.getenv('LDAP_MOBILE_FIELD', 'mobile')
+
+        doc.ssl_tls_mode = os.getenv('LDAP_SSL_TLS_MODE')
+
+        # TODO Enable LDAP Settings
```
Comment (madmath03) [inline erpnext_autoinstall/install.py]: @AminovE99 We need to add a `LDAP_ENABLED` config

**Review by madmath03 — COMMENTED** [review]
Few things yet to improve.

**Pull request merged**